pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+    var ret = [ ];
+    var parts = text.split('\n\n');
+    var i;
+    for (i = 0; i < parts.length; ++i) {
+        var segment = parts[i];
+        if (segment.indexOf('Product Name:') === -1)
+            continue;
+
+        var segmentInfo = parseSingleSubscription(segment);
+
+        var status = segmentInfo['Status'];
+
+        /* if we have status details, add those to the status */
+        if (segmentInfo['Status Details'] !== '')
+            status = status + ' (' + segmentInfo['Status Details'] + ')';
+
+        /* convert text output to mustache template variables */
+        ret.push({
+            'productName': segmentInfo['Product Name'],
+            'productId': segmentInfo['Product ID'],
+            'version': segmentInfo['Version'],
+            'arch': segmentInfo['Arch'],
+            'status': status,
+            'starts': segmentInfo['Starts'],
+            'ends': segmentInfo['Ends'],
+        });
+    }
+    return ret;
+};
+
+var gettingDetails = false;
+var getDetailsRequested = false;
+function getSubscriptionDetails() {
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    if (gettingDetails) {
+        getDetailsRequested = true;
+        return;
+    }
+    getDetailsRequested = false;
+    gettingDetails = true;
+    cockpit.spawn(['subscription-manager', 'list'],
+            { directory: '/', superuser: "try", environ: ['LC_ALL=C'] })
+        .done(function(output) {
+            client.subscriptionStatus.products = parseMultipleSubscriptions(output);
+            gettingDetails = false;
+            if (getDetailsRequested)
+                getSubscriptionDetails();
+            needRender();
+        })
+        .fail(function(ex) {
+            client.subscriptionStatus.error = ex;
+            console.warn("Subscriptions [getSubscriptionDetails]: couldn't get details: " + ex);
+            if (getDetailsRequested)
+                getSubscriptionDetails();
+            needRender();
+        });
+}
+
+client.registerSystem = function(subscriptionDetails) {
+    var deferred = $.Deferred();
+
+    var args = ['subscription-manager', 'register'];
+    if (subscriptionDetails.url != 'default')
+        args.push('--serverurl', subscriptionDetails.serverUrl);
+
+    // activation keys can't be used with auto-attach
+    if (subscriptionDetails.activationKeys)
+        args.push('--activationkey', subscriptionDetails.activationKeys);
+    else
+        args.push('--auto-attach');
+
+    if (subscriptionDetails.user || subscriptionDetails.password)
+        args.push('--username', subscriptionDetails.user, '--password', subscriptionDetails.password);
+
+    // proxy is optional
+    if (subscriptionDetails.proxy) {
+        args.push('--proxy', subscriptionDetails.proxyServer,
+                  '--proxyuser', subscriptionDetails.proxyUser,
+                  '--proxypass', subscriptionDetails.proxyPass);
```

Comment:
Any null or undefined fields need to be collapsed to ''. Maybe that's happening, but can't tell.